store/v2/commitment/iavl/tree.go

```diff
@@ -83,6 +83,16 @@ func (t *IavlTree) Commit() ([]byte, uint64, error) {
 
 // GetProof returns a proof for the given key and version.
 func (t *IavlTree) GetProof(version uint64, key []byte) (*ics23.CommitmentProof, error) {
+	// the mutable tree is empty at genesis & when the key is removed, but the immutable tree is not
+	// by checking the latest version we can determine if we are in genesis or have a key that has been removed
+	lv, err := t.tree.GetLatestVersion()
+	if err != nil {
+		return nil, err
+	}
+	if lv == 0 {
```

Comment:
when starting a node the version of the tree will always be 0, since its empty and if the root is nil the version is assumed to be 0, but when you write (apply change sets) it would be on the height/version you specify. 

If you remove a store key the values are still there, unless pruned. In that case the mutable tree will nil because the version of the tree could be 30 but the key was removed at 20. Clients may still want to query version(s) <20. In this case the latest version of the tree will be 20 not 30.